lib/apt/spy2.rb

```diff
@@ -121,6 +121,42 @@ def filter(mirrors, strict = false, output = true)
 
   end
 
+  private
+  # architecture is not reported in the strings we need
+  def fix_arch(str)
```

Comment:
@parthshyara This by itself made it work, but it was very slow because of the size of the `.gz` files and `URI.open()` is pretty much a `GET` request and reads everything into memory.